### PR TITLE
feat: avm/res/network/route-table

### PR DESCRIPTION
**Pull request by segraef**
[avm.res.network.route-table](https://github.com/segraef/bicep-registry-modules/actions/workflows/avm.res.network.route-table.yml)

### CODE REVIEW

**Review by AlexanderSehr — CHANGES_REQUESTED** [review]

**Review by AlexanderSehr — CHANGES_REQUESTED** [review]
Looks good. Just as a last ask: please attach a pipeline badge. 😉

**Review by skydiver1980 — APPROVED** [review]

File: avm/res/network/route-table/main.bicep [hunk 1]
```diff
@@ -0,0 +1,134 @@
+metadata name = 'Route Tables'
```
Comment (eriqua) [inline avm/res/network/route-table/main.bicep:1]: Please also make sure of the following:
- update codeowners file uncommenting the line referencing your module
- make sure your module owner team is properly added to the intended parent team (cc @matebarabas)

**Comment by segraef** [comment]
> Looks good. Just as a last ask: please attach a pipeline badge. 😉

[avm.res.network.route-table](https://github.com/segraef/bicep-registry-modules/actions/workflows/avm.res.network.route-table.yml)

File: avm/res/network/route-table/main.bicep (hunk 1 above)
Comment (segraef) [reply avm/res/network/route-table/main.bicep:1]: Done, also created both teams

https://github.com/orgs/Azure/teams/avm-res-network-routetable-module-owners-bicep 
https://github.com/orgs/Azure/teams/avm-res-network-routetable-module-contributors-bicep

and added corresponding parent teams as per https://azure.github.io/Azure-Verified-Modules/specs/shared/#grant-permissions---bicep

File: avm/res/network/route-table/main.bicep [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+    level: lock.?kind ?? ''
+    notes: lock.?kind == 'CanNotDelete' ? 'Cannot delete resource or child resources.' : 'Cannot delete or modify the resource or child resources.'
+  }
+  scope: routeTable
+}
+
+resource routeTable_roleAssignments 'Microsoft.Authorization/roleAssignments@2022-04-01' = [for (roleAssignment, index) in (roleAssignments ?? []): {
+  name: guid(routeTable.id, roleAssignment.principalId, roleAssignment.roleDefinitionIdOrName)
+  properties: {
+    roleDefinitionId: contains(builtInRoleNames, roleAssignment.roleDefinitionIdOrName) ? builtInRoleNames[roleAssignment.roleDefinitionIdOrName] : roleAssignment.roleDefinitionIdOrName
```
Comment (AlexanderSehr) [inline avm/res/network/route-table/main.bicep:78]:
```suggestion
    roleDefinitionId: contains(builtInRoleNames, roleAssignment.roleDefinitionIdOrName) ? builtInRoleNames[roleAssignment.roleDefinitionIdOrName] : contains(roleAssignment.roleDefinitionIdOrName, '/providers/Microsoft.Authorization/roleDefinitions/') ? roleAssignment.roleDefinitionIdOrName : subscriptionResourceId('Microsoft.Authorization/roleDefinitions', roleAssignment.roleDefinitionIdOrName)
```

Latest improvements we added to the AVM schema. Requires a regeneration of the files

File: avm/res/network/route-table/tests/e2e/max/main.test.bicep [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+    lock: {
+      kind: 'CanNotDelete'
+      name: 'myCustomLockName'
+    }
+    roleAssignments: [
+      {
+        roleDefinitionIdOrName: 'Reader'
+        principalId: nestedDependencies.outputs.managedIdentityPrincipalId
+        principalType: 'ServicePrincipal'
+      }
```
Comment (AlexanderSehr) [inline avm/res/network/route-table/tests/e2e/max/main.test.bicep:62]:
```suggestion
      {
        roleDefinitionIdOrName: 'Owner'
        principalId: nestedDependencies.outputs.managedIdentityPrincipalId
        principalType: 'ServicePrincipal'
      }
      {
        roleDefinitionIdOrName: 'b24988ac-6180-42a0-ab88-20f7382dd24c'
        principalId: nestedDependencies.outputs.managedIdentityPrincipalId
        principalType: 'ServicePrincipal'
      }
      {
        roleDefinitionIdOrName: subscriptionResourceId('Microsoft.Authorization/roleDefinitions', 'acdd72a7-3385-48ef-bd42-f606fba81ae7')
        principalId: nestedDependencies.outputs.managedIdentityPrincipalId
        principalType: 'ServicePrincipal'
      }
```

To test change in the other suggestion.

Comment (AlexanderSehr) [reply avm/res/network/route-table/tests/e2e/max/main.test.bicep:62]: **Note:** This requries an update of the readme.

**Review by AlexanderSehr — APPROVED** [review]

**Pull request merged**